xonsh/tools.py

```diff
@@ -521,19 +513,16 @@ def always_false(x):
 
 def ensure_string(x):
     """Returns a string if x is not a string, and x if it already is."""
-    if isinstance(x, string_types):
-        return x
-    else:
-        return str(x)
+    return str(x)
```

Comment:
i'd say that originates in Python's way to store values in hashing tables. a long time ago i read an article discussing this, but can't remember what it was.

```
>>> id('x' *10000)
28875296
>>> id(str('x' *10000))
28883376
```

why would a `y` matter here?